lib/src/git_dir.dart

```diff
@@ -69,7 +69,11 @@ class GitDir {
 
     for (var ref in refs) {
       final pr = await runCommand(['cat-file', '-p', ref.sha]);
-      yield Tag.parseCatFile(pr.stdout as String);
+      yield Tag.parseCatFile(
+        pr.stdout as String,
+        ref.sha,
+        ref.reference.substring(10),
```

Comment:
why always the substring here?

Reply:
@wgledbe's changes, but I can look into it more and possible alternatives.

Reply:
Looks like the tag's name starts at the 10th index, example of `ref.reference value`:

> refs/tags/newTag

Reply:
@kevmoo I replaced the substring with a split and last:

>   tag = ref.reference.split('/').last;

And moved it into the parsing method